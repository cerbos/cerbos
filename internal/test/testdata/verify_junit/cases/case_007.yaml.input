-- suite_test.yaml --
---
name: CompanyTestSuite
description: Tests for verifying the company resource policy
principals:
  admin:
    id: admin
    roles:
      - admin
  user:
    id: user
    roles:
      - user
resources:
  company:
    id: xx1
    kind: company

tests:
  - name: Company Test 1
    input:
      principals:
        - admin
      resources:
        - company
      actions:
        - create
        - read
    expected:
      - principal: admin
        resource: company
        actions:
          create: EFFECT_ALLOW
          read: EFFECT_ALLOW
        outputs:
          - action: create
            expected:
              - src: resource.company.vdefault#rule-001
                val: "foo"

  - name: Company Test 2
    input:
      principals:
        - user
      resources:
        - company
      actions:
        - create
        - read
    expected:
      - principal: user
        resource: company
        actions:
          create: EFFECT_ALLOW
          read: EFFECT_ALLOW
        outputs:
          - action: create
            expected:
              - src: resource.company.vdefault#rule-001
                val:
                  principal: user
                  resource: xx1

          - action: read
            expected:
              - src: resource.company.vdefault#rule-001
                val:
                  principal: blah
                  resource: yy1

