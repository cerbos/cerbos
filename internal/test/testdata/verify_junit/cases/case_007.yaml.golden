<testsuites errors="0" failures="2" skipped="0" tests="4">
  <testsuite description="Tests for verifying the company resource policy" name="CompanyTestSuite" file="suite_test.yaml" errors="0" failures="2" skipped="0" tests="4">
    <properties></properties>
    <testcase file="suite_test.yaml" classname="admin.company.create" name="Company Test 1">
      <failure type="RESULT_FAILED" message="Output expectation unsatisfied">
        <outputs>
          <output src="resource.company.vdefault#rule-001">
            <expected><![CDATA["foo"]]></expected>
            <actual></actual>
          </output>
        </outputs>
        <actual>EFFECT_ALLOW</actual>
        <expected>EFFECT_ALLOW</expected>
      </failure>
      <properties>
        <property name="principal">admin</property>
        <property name="resource">company</property>
        <property name="action">create</property>
      </properties>
    </testcase>
    <testcase file="suite_test.yaml" classname="admin.company.read" name="Company Test 1">
      <properties>
        <property name="principal">admin</property>
        <property name="resource">company</property>
        <property name="action">read</property>
      </properties>
    </testcase>
    <testcase file="suite_test.yaml" classname="user.company.create" name="Company Test 2">
      <properties>
        <property name="principal">user</property>
        <property name="resource">company</property>
        <property name="action">create</property>
      </properties>
    </testcase>
    <testcase file="suite_test.yaml" classname="user.company.read" name="Company Test 2">
      <failure type="RESULT_FAILED" message="Output expectation unsatisfied">
        <outputs>
          <output src="resource.company.vdefault#rule-001">
            <expected><![CDATA[{
  "principal":  "blah",
  "resource":  "yy1"
}]]></expected>
            <actual><![CDATA[{
  "principal":  "user",
  "resource":  "xx1"
}]]></actual>
          </output>
        </outputs>
        <actual>EFFECT_ALLOW</actual>
        <expected>EFFECT_ALLOW</expected>
      </failure>
      <properties>
        <property name="principal">user</property>
        <property name="resource">company</property>
        <property name="action">read</property>
      </properties>
    </testcase>
  </testsuite>
</testsuites>